docs/DeveloperGuide.md

```diff
@@ -383,27 +358,29 @@ A similar execution scenario can be expected for view lesson mechanism.
     
 ### Card-like UI Elements
 
-Card-like UI elements are objects that are shown to the user in their respective list panels, such as `StudentCard` which is displayed in the `StudentListPanel`. These cards come in two flavours: a fully-detailed variant and a minimally-detailed variant. The fully-detailed variant shows all properties while the minimally-detailed variant keeps the list compact and allows the user to view more entries. 
+Card-like UI elements are objects that are shown to the user in their respective list panels, such as `StudentCard` which is displayed in the `StudentListPanel`. The Student Cards come in three flavours: `FullStudentCard` displays all fields to the user, `StudentCard` is the same but only displays the most recent `Progress` entry, and `MinimalStudentCard` only displays the index. Lesson Cards come in 2 flavours: `LessonCard` (fully-detailed) and `MinimalLessonCard` (only index and timing). Having minimally-detailed variants keeps the list compact and allows the user to view more entries. 
 
 These UI elements inherit the `Card` class, which in turn inherits `UiPart<Region>`. 
 
 ![CardClassDiagram](images/CardClassDiagram.png)
 
 At all times, the `LessonListPanel` and `StudentListPanel` in the `MainWindow` will display Lessons and Students from the model using either the fully-detailed or minimal `Card` objects. The variant being displayed depends on the user command: `list -a` will cause both panels to display all details while `list` will cause both panels to display only minimal details. Most other commands that affect the `Model` will cause all information to be displayed.
```

Comment:
May have to update this paragraph, since I think `list -a` does not show all progress entries for the students, but it's stated here that it will cause the student panel to display all details